File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n = nums.length;
        int sum = 0;
        int ans = Integer.MIN_VALUE;
        for (int i = 0; i < n; i++) {
            sum += nums[i];
            if (sum > ans) {
                ans = sum;
            }
            if (sum < 0) {
                sum = 0;
            }
        }
        return ans;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
   int maxSubArray(vector<int>& nums) {
         int n = nums.size(), sum = 0, max_sum = nums[0];
         for (int i = 0; i < n; i++) {
             sum += nums[i];
             max_sum = max(max_sum, sum);
             if (sum < 0) sum = 0;
         }
         return max_sum;
     }
     
};